### PR TITLE
fix(core): Convert authentication config source functions to actual functions

### PR DESCRIPTION
**Pull request by FokkeZB**
`IP-127` [Zapier Convert: Convert auth config source to functions](https://zapierorg.atlassian.net/browse/IP-127)

Inline functions under auth config objects are not converted to functions by `zapier convert`, while the test function is:

```js
const testAuth = async (z, bundle) => {
  z.request({
    url: "https://api.wistia.com/v1/account.json",
    method: "GET",
    headers: {
      Authorization: "Bearer {{bundle.authData.access_token}}",
    },
    body: {},
  }).then((response) => response.data);
};

module.exports = {
  type: 'oauth2',
  test: testAuth,
  fields: [],
  oauth2Config: {
    getAccessToken: {
      source:
        "const options = {\n  url: 'https://api.wistia.com/oauth/token',\n  method: 'POST',\n  headers: {\n    'content-type': 'application/x-www-form-urlencoded',\n    accept: 'application/json',\n  },\n  body: {\n    grant_type: 'refresh_token',\n    refresh_token: '{{bundle.authData.refresh_token}}',\n  }\n}\n\nreturn z.request(options)\n  .then((response) => {\n    const results = response.data;\n\n    // You can do any parsing you need for results here before returning them\n\n    return results;\n  });",
    },
  },
};
```

This PR fixes that my traversing to look for any `source:` and adds missing test coverage.

### CODE REVIEW

**Comment by FokkeZB** [comment]
@eliangcs this change is likely less controversial and will help with Py2CLI.

**Review by eliangcs — DISMISSED** [review]
I tested with a few apps and it works great! I only have a comment on the traverse package version used in `package.json`. Otherwise, it looks good to me!

File: packages/cli/package.json [hunk 1]
```diff
@@ -68,6 +68,7 @@
     "string-length": "4.0.2",
     "through2": "4.0.2",
     "tmp": "0.2.1",
+    "traverse": "^0.6.7",
```
[the remark this answers is not in the document]
Comment (FokkeZB) [reply packages/cli/package.json]: Done!

**Review by eliangcs — APPROVED** [review]
Thanks! Merging.

**Pull request merged**